java/ql/test/kotlin/library-tests/parameter-defaults/PrintAst.expected

```diff
@@ -0,0 +1,849 @@
+#   74|       5: [BlockStmt] { ... }
+#   75|         0: [ExprStmt] <Expr>;
+#   75|           0: [ImplicitCoercionToUnitExpr] <implicit coercion to unit>
+#   75|             0: [TypeAccess] Unit
+#   75|             1: [ClassInstanceExpr] new TestConstructor(...)
+#   75|               -3: [TypeAccess] TestConstructor
+#   75|               0: [StringLiteral] constructor sunk
+#    1|               1: [NullLiteral] null
+#    1|               2: [NullLiteral] null
+#    1|               3: [IntegerLiteral] 1
+#    1|               4: [NullLiteral] null
+#   76|         1: [ExprStmt] <Expr>;
+#   76|           0: [ImplicitCoercionToUnitExpr] <implicit coercion to unit>
+#   76|             0: [TypeAccess] Unit
+#   76|             1: [ClassInstanceExpr] new TestConstructor(...)
+#   76|               -3: [TypeAccess] TestConstructor
+#   76|               0: [StringLiteral] constructor sunk fp
+#   76|               1: [StringLiteral] constructor sunk 2
+#    1|               2: [NullLiteral] null
+#    1|               3: [IntegerLiteral] 3
+#    1|               4: [NullLiteral] null
+#   77|         2: [ExprStmt] <Expr>;
+#   77|           0: [ImplicitCoercionToUnitExpr] <implicit coercion to unit>
+#   77|             0: [TypeAccess] Unit
+#   77|             1: [ClassInstanceExpr] new TestConstructor(...)
+#   77|               -3: [TypeAccess] TestConstructor
+#   77|               0: [StringLiteral] not sunk
+#   77|               1: [StringLiteral] constructor sunk 3
+#   77|               2: [StringLiteral] not sunk
+#   82|   8: [Class] TestLocal
+#   82|     1: [Constructor] TestLocal
+#   82|       5: [BlockStmt] { ... }
+#   82|         0: [SuperConstructorInvocationStmt] super(...)
+#   82|         1: [BlockStmt] { ... }
+#   84|     2: [Method] enclosing
+#   84|       3: [TypeAccess] Unit
+#   84|       5: [BlockStmt] { ... }
+#   86|         0: [LocalTypeDeclStmt] class ...
+#   86|           0: [LocalClass] 
+#   86|             1: [Constructor] 
+#   86|               5: [BlockStmt] { ... }
+#   86|                 0: [SuperConstructorInvocationStmt] super(...)
+#   86|             2: [Method] f
+#   86|               3: [TypeAccess] Unit
+#-----|               4: (Parameters)
+#   86|                 0: [Parameter] x
+#   86|                   0: [TypeAccess] String
+#   86|                 1: [Parameter] y
+#   86|                   0: [TypeAccess] String
+#   86|                 2: [Parameter] z
+#   86|                   0: [TypeAccess] String
+#   86|               5: [BlockStmt] { ... }
+#   87|                 0: [ExprStmt] <Expr>;
+#   87|                   0: [MethodAccess] sink(...)
+#   87|                     -1: [TypeAccess] TestKt
+#   87|                     0: [VarAccess] y
+#   86|             3: [Method] f$default
+#   86|               3: [TypeAccess] Unit
+#-----|               4: (Parameters)
+#   86|                 0: [Parameter] p0
+#   86|                   0: [TypeAccess] String
```

Comment:
Shouldn't the first parameter be the dispatch receiver?

Reply:
Not for a local function it seems:

```
public class Test {
  fun f() {
    fun g(x: Int = 1) { }
  }
}
```

Leads to

```
  private static final void f$g(int);
...
  static void f$g$default(int, int, java.lang.Object);
```

Reply:
Hmm, it might be somehow compiler or target version dependent. The byte code viewer in IntelliJ shows me the following. (I stripped the noise for brevity.)
```
final class Test$f$1 extends kotlin/jvm/internal/Lambda implements kotlin/jvm/functions/Function1 {
  public final invoke(I)V
  public static synthetic invoke$default(LTest$f$1;IILjava/lang/Object;)V
}
```

Reply:
@smowton I wonder if this difference can only happen for constructs that are definitely not visible from outside. Otherwise it could occur that we're not compatible with the java extractor, couldn't it?

Reply:
Hmm, true. I'm inclined to punt on the details for now on the grounds that function-local declarations getting direct calls from Java is surely very niche.